model-optimizer/mo/back/preprocessing.py

```diff
@@ -0,0 +1,383 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import logging as log
+
+from mo.utils.error import Error
+from mo.utils.utils import refer_to_faq_msg
+
+import numpy as np
+
+from openvino.preprocess import PrePostProcessor        # pylint: disable=no-name-in-module,import-error
+# pylint: disable=no-name-in-module,import-error
+from openvino.runtime import Function, Layout, PartialShape, layout_helpers
+
+
+def update_mean_scale_to_dict(input_nodes: list, mean_scale_val, scale):
+    """
+    Internal function. Updates mean/scale values from array to dictionary
+    :param: input_nodes Inputs of model
+    :param: mean_scale_val Parsed 'mean_scale_val' object from command line arguments
+    :param: scale Global scale factor for all inputs from --scale command line arguments
+    """
+    if not isinstance(mean_scale_val, dict):
+        if len(mean_scale_val) != len(input_nodes):
+            raise Error('Numbers of inputs and mean/scale values do not match. ' + refer_to_faq_msg(61))
+
+        data = np.copy(mean_scale_val)
+        mean_scale_val = {}
```

Comment:
please add assert to check len(input_nodes) == len(mean_scale_val)

Reply:
hm, this check is 2 lines above...

Reply:
yeah, you are correct